src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -1153,10 +1153,14 @@ private Selection prepareSelection(TableMetadata table,
             if (hasGroupBy)
                 Guardrails.groupByEnabled.ensureEnabled(state);
 
+            boolean isJson = parameters.isJson;
+            boolean returnStaticContentOnPartitionWithNoRows = restrictions.returnStaticContentOnPartitionWithNoRows();
+
             if (selectables.isEmpty()) // wildcard query
             {
-                return hasGroupBy ? Selection.wildcardWithGroupBy(table, boundNames, parameters.isJson, restrictions.returnStaticContentOnPartitionWithNoRows())
-                                  : Selection.wildcard(table, parameters.isJson, restrictions.returnStaticContentOnPartitionWithNoRows());
+                return hasGroupBy || table.columns().stream().anyMatch(ColumnMetadata::isMasked)
```

Comment:
Avoid streams?

Reply:
Same as before, it's not in the hot path.

Reply:
Added that convenience `TableMetadata#hasMaskedColumns()` method, which is way cleaner. Since that's now a general purpose method that might called in any path I'm changing it to a loop.